code/controllers/subsystems/initialization/materials.dm

```diff
@@ -58,8 +58,9 @@ SUBSYSTEM_DEF(materials)
 				alloy_components[component] = TRUE
 
 /datum/controller/subsystem/materials/proc/get_material_datum(var/mat)
-	. = materials_by_name[mat]
-	if(!.)
+	try
```

Comment:
This is unnecessary. `materials_by_name` is always a list, `.` will either be a material or null.

Reply:
accessing materials_by_name by an invalid mat will throw a runtime. Bad index runtimes exist, it needs to be in a try catch.

Reply:
Bad index runtime only occurs if it's a number, in which case that runtime is already a sufficient way to handle the situation.

Reply:
When I encountered the runtime, it was insufficient information to debug. The more verbose way is the more than sufficient way to handle it imo

Reply:
How is it insufficient? The example you showed was a snippet from DD after it had been called so often that DD started collapsing the runtimes. The information is still in the logs.

Reply:
Your assumption is incorrect. That error only appeared because of my code.